Source/Core/sampleTerrain.js

```diff
@@ -48,6 +48,78 @@ function sampleTerrain(terrainProvider, level, positions) {
   });
 }
 
+const ACTION__CONTINUE = "continue";
+const ACTION__DELAY = "delay";
+
+function consumeQueueItem(tileRequests, inflightRequests) {
+  const tileRequest = tileRequests[0];
+  const requestPromise = tileRequest.terrainProvider.requestTileGeometry(
+    tileRequest.x,
+    tileRequest.y,
+    tileRequest.level
+  );
+
+  if (!requestPromise) {
+    // Uh, we probably got throttled client side?
+    //  let's wait 100ms, then try again
+    // This will turn into an infinite loop if we don't eventually
```

Comment:
This is true of other areas of the code which use the request scheduler. I wouldn't be too concerned, as the only situation which this returns undefined should be one where a retry is needed.